android/app/src/main/java/com/expensify/chat/bootsplash/BootSplashActivity.java

```diff
@@ -0,0 +1,42 @@
+package com.expensify.chat.bootsplash;
+
+import android.content.Intent;
+import android.os.Bundle;
+
+import androidx.annotation.Nullable;
+import androidx.appcompat.app.AppCompatActivity;
+
+import com.expensify.chat.MainActivity;
+
+public class BootSplashActivity extends AppCompatActivity {
+
+  protected void forwardIntentToMainActivity(Intent intent) {
+    Intent intentCopy = new Intent(intent);
+
+    intentCopy.setClass(this, MainActivity.class);
+    intentCopy.putExtras(intent);
+    intentCopy.setData(intent.getData());
+    intentCopy.setAction(intent.getAction());
+
+    String type = intent.getType();
+
+    if (type != null) {
+      intentCopy.setType(type);
+    }
+
+    startActivity(intentCopy);
+    finish();
+  }
+
+  @Override
+  protected void onNewIntent(Intent intent) {
+    super.onNewIntent(intent);
+    forwardIntentToMainActivity(intent);
+  }
+
+  @Override
+  protected void onCreate(@Nullable Bundle savedInstanceState) {
+    super.onCreate(savedInstanceState);
+    forwardIntentToMainActivity(getIntent());
+  }
```

Comment:
This is super minor and not a blocker, but could we keep `onCreate` as the topmost function for consistency? Thanks.

Reply:
Additionally, could we list the overridden functions above `forwardIntentToMainActivity`.